bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller_test.go

```diff
@@ -1070,18 +1070,52 @@ func TestBootstrapTokenTTLExtension(t *testing.T) {
 	g.Expect(cfg.Status.ObservedGeneration).NotTo(BeNil())
 
 	l := &corev1.SecretList{}
-	err = myclient.List(ctx, l, client.ListOption(client.InNamespace(metav1.NamespaceSystem)))
-	g.Expect(err).ToNot(HaveOccurred())
-	g.Expect(l.Items).To(HaveLen(2))
+	g.Expect(remoteClient.List(ctx, l, client.ListOption(client.InNamespace(metav1.NamespaceSystem)))).To(Succeed())
+	g.Expect(l.Items).To(HaveLen(2)) // control plane vs. worker
 
-	// ensure that the token is refreshed...
+	t.Log("Ensure that the token secret is not updated while it's still fresh")
 	tokenExpires := make([][]byte, len(l.Items))
 
 	for i, item := range l.Items {
 		tokenExpires[i] = item.Data[bootstrapapi.BootstrapTokenExpirationKey]
 	}
 
-	<-time.After(1 * time.Second)
```

Comment:
Instead of having tests sleep, which is a bad developer experience, I'm now simulating close-to-expired tokens.